internal/pkg/service/stream/sink/type/tablesink/keboola/bridge/bridge.go

```diff
@@ -105,10 +105,10 @@ func (b *Bridge) isKeboolaStagingFile(file *model.File) bool {
 	return file.StagingStorage.Provider == stagingFileProvider
 }
 
-func (b *Bridge) MirrorJobs(d dependencies) error {
+func (b *Bridge) MirrorJobs(ctx context.Context, d dependencies) error {
 	// Mirror jobs
 	wg := &sync.WaitGroup{}
-	ctx, cancel := context.WithCancel(context.Background())
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
Shouldn't this be like this? :thinking: I'm not sure because context cancellation is a topic I'm not very experienced with.

```go
ctx, cancel := context.WithCancel(context.WithoutCancel(ctx))
```

Reply:
Once the `coordinator` node is canceled, you want to tear down the `mirror` in the bridge. It does not have totally separate livespan. Using the `coordinator` context, once it is canceled you can teardown the bridge `mirror`. You do not need to wait for termination. It can be done upfront.

If you use thing that you post, the `mirror` can be canceled only when process shutdown signal is received even though rest of program is already in teardown state . Either way we miss the dependency that the `mirror` should be canceled before the `coordinator`.

If you would like to have dependency between these two instances who is served when it has to be done somehow else. Either way it is not bulletproof that the `mirror` is stopped or `mirror` is updated after `coordinator` teardown

Reply:
I see. Thanks for explaining.